lib/server-config-manager.ts

```diff
@@ -23,16 +23,16 @@ export class ServerConfigManager implements IServerConfigManager {
 		this.applyConfig("production");
 	}
 
-	public applyConfig(configName: string, options?: IConfigOptions): void {
+	public applyConfig(configName: string, options?: IConfigOptions, globalServerConfig?: IServerConfigBase): void {
 		const baseConfig = this.loadConfig(ServerConfigManager.BASE_CONFIG_FILE_NAME);
 		const newConfig = this.loadConfig(`${ServerConfigManager.CONFIG_FILE_NAME}-${configName}`, options);
-		this.mergeConfig(baseConfig, newConfig);
+		this.mergeConfig(baseConfig, newConfig, globalServerConfig);
 		this.saveConfig(baseConfig, ServerConfigManager.CONFIG_FILE_NAME);
 	}
 
 	public printConfigData(): void {
 		const config = this.getCurrentConfigData();
-		console.log(config);
+		console.log(JSON.stringify(config, null, "  "));
```

Comment:
Another personal preference - I'd prefer a number as the third argument of `JSON.stringify` - seems a bit more natural to me:
`console.log(JSON.stringify(config, null, 2));`